transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultPersistor.java

```diff
@@ -339,6 +340,32 @@ public Collection<TransactionOutboxEntry> selectNextInTopics(
     }
   }
 
+  @Override
+  public Collection<TransactionOutboxEntry> selectNextInSelectedTopics(
+      Transaction tx, Collection<String> topicNames, int batchSize, Instant now) throws Exception {
```

Comment:
With this code you should enforce an _ordered_ collection so I would require a `List`.

Reply:
true

Reply:
it is an array list under the hood btw - but we can make that explicit on the return type of list instead of collection
> var results = new ArrayList<TransactionOutboxEntry>();

Reply:
sure. I think I am a bit confused then (might be missing the obvious!). 

assuming this is about changing Collection<String> topicNames to List<String> topicNames and propagating this all the way up the call hierarchy. 

Q: why do we need to guarantee ordering between the topics? 
this select grabs only the very next item to be processed for each topic. these topics are independent of one another. I would assume then that the order they get processed in exactly doesn't matter.

Reply:
I see, so this is about desiring predictability in the data that we pass in.
I will make the change, however, this data is being used to populate an IN clause under the hood, within a sql query that does not declare an absolute ordering on the results it returns. We make no promises then on a guaranteed result returned from this method.

Reply:
Fair point actually, since `?`s don't need to be in any particular order! Sorry I'm being thick.